Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -1055,20 +1052,29 @@ - (void)initializeSession {
 }
 
 - (void)registerInstallOrOpen:(Class)clazz {
+    callbackWithStatus initSessionCallback = ^(BOOL success, NSError *error) {
+        if (error) {
+            [self handleInitFailure:error];
+        }
+        else {
+            [self handleInitSuccess];
+        }
+    };
+
+    // If there isn't already an Open / Install request, add one to the queue
     if (![self.requestQueue containsInstallOrOpen]) {
-        BranchOpenRequest *req = [[clazz alloc] initWithCallback:^(BOOL success, NSError *error) {
-            if (error) {
-                [self handleInitFailure:error];
-            }
-            else {
-                [self handleInitSuccess];
-            }
-        }];
+        BranchOpenRequest *req = [[clazz alloc] initWithCallback:initSessionCallback];
 
         [self insertRequestAtFront:req];
     }
+    // If there is already one in the queue, make sure it's in the front.
+    // Make sure a callback is associated with this request. This callback can
+    // be cleared if the app is terminated while an Open/Install is pending.
     else {
         [self.requestQueue moveInstallOrOpenToFront:self.networkCount];
```

Comment:
it'd be nice for moveInstallOrOpenToFront to return the req after the move directly, such that you don't need a peek after this to get the req.